### PR TITLE
[BC5] Add isPersonalizedPrice to purchaseProduct and purchasePackage

### PR DESCRIPTION
**Pull request by joshdholtz**
## Motivation

[CF-1266](https://revenuecats.atlassian.net/browse/CF-1266)

## Description

Adds `isPersonalizedPrice: Boolean?` argument onto `purchaseProduct` and `purchasePackage`

[CF-1266]: https://revenuecats.atlassian.net/browse/CF-1266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by NachoSoto — APPROVED** [review]
Looks good.

File: android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt [hunk 1]
```diff
@@ -58,6 +58,7 @@ fun purchaseProduct(
     oldSku: String?,
     prorationMode: GoogleProrationMode?,
     type: String,
+    isPersonalizedPrice: Boolean?,
```
Comment (beylmk) [inline android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt:61]: i've toyed in my head with whether isPlayPricePersonalized or something that indicates this is google-specific would be better naming... and it feels even more important maybe in the hybrids. thoughts?

Comment (beylmk) [reply android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt:61]: might end up wanting to do oldGoogleProductId and googldProrationMode too... 

also, speaking of... looks like oldsku can be changed to oldProductId. ignore if you're doing that in the future..

**Review by beylmk — APPROVED** [review]
just looking for feedback on the naming, nothing blocking

**Pull request merged**